tools/eslint-rules/only-ascii-characters.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @fileoverview Prohibit the use of non-ascii characters
+ * @author Kalon Hinds
+ */
+
+/* eslint no-control-regex:0 */
+
+'use strict';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+const nonAsciiPattern = new RegExp('([^\x00-\x7F])', 'g');
+const suggestions = {
+  '’': '\'',
+  '—': '-'
+};
+
+module.exports = {
+  create: function(context) {
+    return {
+      Program: (node) => {
+        const errors = [];
+        const source = context.getSourceCode();
+        const sourceTokens = source.getTokens(node);
+        const commentTokens = source.getAllComments();
+        const tokens = sourceTokens.concat(commentTokens);
+
+        tokens.forEach((token) => {
+          const { value } = token;
+          const matches = value.match(nonAsciiPattern);
+
+          if (matches) {
+            const { loc } = token;
+
+            // Will only report the first non-ascii character per line
+            const character = matches[0];
+
+            errors.push({
+              line: loc.start.line,
+              column: loc.start.column + value.indexOf(character),
+              character
+            });
+          }
+        });
+
+        errors.forEach((error) => {
```

Comment:
Why not do this in one pass without the extra `errors` array? You can make this a named function and call it instead of `errors.push()`.